bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/ThermostatHandler.java

```diff
@@ -197,7 +197,7 @@ private void updateManualSetpoint(ThermostatModel thermostat) {
     }
 
     private void updateManualSetpoint(Command command) {
-        if (command instanceof QuantityType<?> quantityCommand) {
+        if (command instanceof Number quantityCommand) {
```

Comment:
Rather support the two options, Number and QuantityType.

Reply:
I think QuantityType extends Number, so I think it should work, right?

[https://www.openhab.org/javadoc/latest/org/openhab/core/library/types/quantitytype](https://www.openhab.org/javadoc/latest/org/openhab/core/library/types/quantitytype)

Reply:
In fact, I was thinking about `DecimalType `vs `QuantityType`. Yes, both are `Number`.
What is expected is that you can accept any temperature whatever its unit; that means values in ° Celsius or in ° Fahrenheit. So if you have a QuantityType as command value, you should first convert the value to the unit you finally expect (maybe Celsius, I don't know if this is the expected unit).
If you have no unit in the value you receive as command, you can consider a default unit of course.
Sorry, my comment was not very precise and not enough clear.

What is the unit expected for `getCurrentThermostat().manualModeSetpoint `?

Reply:
Now i got it. The target value is °C. So if the command is QuantityType i convert it to °C and if it's a decimal then i accept it always as °C. Is this the solution?

Reply:
Exactly. You have many examples in other bindings implementing a set point channel.
Let me know if you need help to find such an example.

Reply:
I hope if found a better solution now. So please have a look again @lolodomo

Reply:
That is OK